src/cargo-kani/src/args.rs

```diff
@@ -0,0 +1,313 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::bail;
+use clap::{arg_enum, Error, ErrorKind};
+use std::ffi::OsString;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "kani",
+    about = "Verify a single Rust file. For more information, see https://github.com/model-checking/kani"
+)]
+pub struct StandaloneArgs {
+    /// Rust file to verify
+    #[structopt(parse(from_os_str))]
+    pub input: PathBuf,
+
+    #[structopt(flatten)]
+    pub common_opts: KaniArgs,
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "cargo-kani",
+    about = "Verify a Rust crate. For more information, see https://github.com/model-checking/kani"
+)]
+pub struct CargoKaniArgs {
+    #[structopt(flatten)]
+    pub common_opts: KaniArgs,
+}
+
+// Common arguments for invoking Kani. This gets put into KaniContext, whereas
+// anything above is "local" to "main"'s control flow.
+#[derive(Debug, StructOpt)]
+pub struct KaniArgs {
+    /// Generate visualizer report to <target-dir>/report/html/index.html
+    #[structopt(long)]
+    pub visualize: bool,
+    /// Keep temporary files generated throughout Kani process
+    #[structopt(long)]
+    pub keep_temps: bool,
+
+    /// Produce full debug information
+    #[structopt(long)]
+    pub debug: bool,
+    /// Produces no output, just an exit code and requested artifacts; overrides --verbose
+    #[structopt(long, short)]
+    pub quiet: bool,
+    /// Output processing stages and commands, along with minor debug information
+    #[structopt(long, short)]
+    pub verbose: bool,
+    /// Print commands instead of running them
+    #[structopt(long)]
+    pub dry_run: bool,
+    /// Generate C file equivalent to inputted program
+    #[structopt(long)]
+    pub gen_c: bool,
+
+    // TODO: currently only cargo-kani pays attention to this.
+    /// Directory for all generated artifacts
+    #[structopt(long, parse(from_os_str))]
+    pub target_dir: Option<PathBuf>,
```

Comment:
Is it possible to set a default value here to avoid hard coding the default value in other places?

Reply:
No: the default is different for standalone/cargo. The "hard coding in other places" is just the two different defaults.

I agree there's a design problem there, but it's not an obvious one to fix. I've deliberately avoided letting a `KaniSession` know if it's `cargo-kani` or `kani` because it's just supposed to represent the flags, but here's an area where it matters, and it gets a little annoying.